pkg/operatingsystem/kernel-package.go

```diff
@@ -46,6 +46,23 @@ func (kp *KernelPackage) ProbeName() string {
 	return fmt.Sprintf("%s_%s_%s_%s", driverName, targetID, kernelRelease, kernelVersion)
 }
 
+// KernelPackageFromProbeName extracts the kernel package from a full probe name
+func KernelPackageFromProbeName(probe string) string {
```

Comment:
This will annoyingly only work for Operating Systems which expose the kernel release information as part of their package names :( .
When we get onto other operating systems which don't have package managers e.g. GKE (`cos`), we'll be using Build IDs as Kernel Package Names (https://github.com/falcosecurity/falco/blob/0.29.1/scripts/falco-driver-loader#L327). We can list available Build IDs (Kernel Packages) with ```gsutil ls -r gs://cos-tools/ | grep "kernel-headers.tgz$"```

Reply:
Let me see if I get this now:

At build time, we only have the Build ID.
Once the probe is built, we only have the kernel package name (embedded in the probe name).

And there is no way to go from one to the other without; having a hard-coded list/doing some network request type stuff.

If that's the case, can we consider supporting that work to be out-of-scope of this change, keeping the first version using the probe name, knowing we'll need to update it for COS?

Reply:
Close, I think just the terminology is a bit out of order.

At build time, we only have the _Kernel Package Name_, (`plz run //cmd/build-falco-ebpf-probe -- <operating system> <kernel package name>` ). 

For example, with `amazonlinux2` (EKS):
```bash
# 1. list available kernel packages
$ plz run //cmd/list-kernel-packages -- amazonlinux2
...
4.14.77-86.82.amzn2
...
# 2. build probe for chosen kernel package
$ plz run //cmd/build-falco-ebpf-probe -- amazonlinux2 4.14.77-86.82.amzn2
# outputs falco_amazonlinux2_4.14.77-86.82.amzn2.x86_64_1.o
```

For GKE `cos`, we'll be using the _Build ID_ as the _Kernel Package Name_ as there is no package manager for `cos`. e.g.
```bash
# 1. list available kernel packages
$ plz run //cmd/list-kernel-packages -- cos
# this will use the output of `gsutil ls -r gs://cos-tools/ | grep "kernel-headers.tgz$" | cut -f4 -d/`
...
16108.470.11
...
# 2. build probe for chosen kernel package
$ plz run //cmd/build-falco-ebpf-probe -- cos 16108.470.11
# outputs something like falco_cos_4.14.77-86.82.x86_64_20.o
```

So it's impossible for us to convert a `cos` _Kernel Package Name_ into a Falco probe filename and vice-versa without a hardcoded mapping somewhere which we could whack in the release notes 😉  as:
| Operating System | Kernel Package Name | Probe Filename |
|----|-------------------------|-------------------|
| `cos` | `16108.470.11` | `falco_cos_4.14.77-86.82.x86_64_20.o` |

There's a few other fiddly things with `cos` too (e.g. different Kernel Packages may produce the same falco probe as they have the same kernel) 😅 Let's mark `cos` out of scope as you've suggested and we can unpack when we get round to it  🚀